### PR TITLE
Fixes #1882: NumericInput tests are shifted to a new folder and their package is updated to match the directory structure.

### LINKED ISSUE #1882: Shift NumericInput Rules test to separate sub-package
`NumericInputEqualsRuleClassifierProviderTest.kt`
`NumericInputIsLessThanRuleClassifierProviderTest.kt`
`NumericInputIsLessThanOrEqualToRuleClassifierProviderTest.kt`
`NumericInputIsInclusivelyBetweenRuleClassifierProviderTest.kt`
`NumericInputIsGreaterThanRuleClassifierProviderTest.kt`
`NumericInputIsGreaterThanOrEqualToRuleClassifierProviderTest.kt`

Shift all these to separate package named `numberinput` in https://github.com/oppia/oppia-android/tree/develop/domain/src/test/java/org/oppia/android/domain/classify/rules

### PR DESCRIPTION
**Pull request by soamOne**
## Explanation

  Fixes #1882: In this PR we shift the numericinput test files to a new folder called numericinput. The package in the files is 
  updated to match the directory structure of the files. The Bazel files are also updated 
  with the correct directory structure.
  

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by soamOne** [comment]
@BenHenning @rt4914 PTAL.

**Comment by soamOne** [comment]
Hey @rt4914 @BenHenning, Why are the Kotlin tests failing due to import statements, even though the last PR passed these tests?

**Comment by aggarwalpulkit596** [comment]
@soamOne the lints are failing. Could you fix those ? Also you can check locally by running ktlint --android

**Comment by soamOne** [comment]
> Hey @rt4914 @BenHenning, Why are the Kotlin tests failing due to import statements, even though the last PR passed these tests?

@aggarwalpulkit596 can you please help with this?

**Comment by anandwana001** [comment]
Hi @soamOne 

Ktlint is failing because you have a few unused imports in these files. I had mentioned the files and the line number, you can check and remove these lines which are unused. Also, remember while removing that the imports should be in lexicographical order else ktlint fail again. To automate this lexicographical order you can reformat the code from the menu option. 

```
numericinput/NumericInputEqualsRuleClassifierProviderTest.kt:13:1: Unnecessary import
numericinput/NumericInputIsGreaterThanOrEqualToRuleClassifierProviderTest.kt:13:1: Unnecessary import
numericinput/NumericInputIsGreaterThanRuleClassifierProviderTest.kt:13:1: Unnecessary import
numericinput/NumericInputIsInclusivelyBetweenRuleClassifierProviderTest.kt:13:1: Unnecessary import
numericinput/NumericInputIsLessThanOrEqualToRuleClassifierProviderTest.kt:13:1: Unnecessary import
numericinput/NumericInputIsLessThanRuleClassifierProviderTest.kt:13:1: Unnecessary import
```

You can simply remove them using optimize imports. Also, for the future works, I suggest you set macro as per description here https://github.com/oppia/oppia-android/wiki/Ktlint-Guide#macros

This will reformat your code and also optimize the imports, so there will be few chances of ktlint fails. 
You can also run ktlint on your local machine as per the instruction here  https://github.com/oppia/oppia-android/wiki/Ktlint-Guide#commands

**Comment by BenHenning** [comment]
You can also run the ktlint tool on the command line with '-F' to automatically fix linter errors, though it's good practice to make sure that you know how to manually fix them as well. :)

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @soamOne! The changes generally look good, but please fix the ktlint errors and I'll take one more pass on this PR.

File: app/src/main/java/org/oppia/android/app/profile/ProfileInputView.kt [hunk 1]
```diff
@@ -60,9 +60,9 @@ class ProfileInputView @JvmOverloads constructor(
     orientation = VERTICAL
     if (
       attributes.getBoolean(
-        R.styleable.ProfileInputView_isPasswordInput,
-        /** defVal= */ false
-      )
+          R.styleable.ProfileInputView_isPasswordInput,
```
Comment (prayutsu) [inline app/src/main/java/org/oppia/android/app/profile/ProfileInputView.kt]: @soamOne You have changed indentation here, fixing that would solve the ktlint error

File: app/src/main/java/org/oppia/android/app/recyclerview/StartSnapHelper.kt [hunk 2]
```diff
@@ -68,8 +68,8 @@ class StartSnapHelper : LinearSnapHelper() {
       return if (helper.getDecoratedEnd(child) >=
         helper.getDecoratedMeasurement(child) / 2 &&
         helper.getDecoratedEnd(
-          child
-        ) > 0
+            child
```
Comment (prayutsu) [inline app/src/main/java/org/oppia/android/app/recyclerview/StartSnapHelper.kt]: Same applies here

**Comment by soamOne** [comment]
Thanks a lot @rt4914 @aggarwalpulkit596 @anandwana001 @prayutsu !!
@BenHenning  can you please take a look now, thanks!

File: app/src/main/java/org/oppia/android/app/profile/ProfileInputView.kt [hunk 3]
```diff
@@ -61,7 +61,8 @@ class ProfileInputView @JvmOverloads constructor(
     if (
       attributes.getBoolean(
         R.styleable.ProfileInputView_isPasswordInput,
-        /** defVal= */ false
+        /** defVal= */
+        false
```
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/profile/ProfileInputView.kt]: One last thing from my side, could you make these in one line?

**Review by anandwana001 — CHANGES_REQUESTED** [review]
@soamOne thanks
LGTM, one last thing.

File: app/src/main/java/org/oppia/android/app/profile/ProfileInputView.kt (hunk 3 above)
Comment (soamOne) [reply app/src/main/java/org/oppia/android/app/profile/ProfileInputView.kt]: Sure, will do!

Comment (soamOne) [reply app/src/main/java/org/oppia/android/app/profile/ProfileInputView.kt]: Done!

**Review by prayutsu — APPROVED** [review]
@soamOne Looks good to me, thanks!

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @soamOne

File: app/src/main/java/org/oppia/android/app/profile/ProfileInputView.kt (hunk 3 above)
Comment (BenHenning) [reply app/src/main/java/org/oppia/android/app/profile/ProfileInputView.kt]: This doesn't appear done yet--please fix.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @soamOne! The PR LGTM, but please fix the extra change in ProfileInputView.kt.

**Comment by soamOne** [comment]
@BenHenning  i tried fixing that but for some reason the checks were failing, so i reverted it.

**Comment by BenHenning** [comment]
@soamOne what specific check was failing? It's important to make sure we don't undo work that we previously mentioned was completed, and to be specific about why changes can't be done otherwise reviewers don't have sufficient context to review the code. :)

**Comment by soamOne** [comment]
@BenHenning I get your point :)
Actually the app module Robolectric checks were failing with a java.lang.RuntimeException.

**Comment by BenHenning** [comment]
Thanks @soamOne! Feel free to ask for help if needed  (e.g. if an error that you encounter doesn't make sense).

**Comment by BenHenning** [comment]
Also, when you encounter an exception you should provide:
- The steps you did to trigger it
- The stack trace you’re seeing
- Your initial thoughts of what’s going wrong and what you’ve tried to fix

Without this information, it’s hard for others to help since they don’t have as much context as you on the inner workings of your PR.

**Comment by soamOne** [comment]
@BenHenning can you PTAL now, thanks :)

**Review by BenHenning — APPROVED** [review]
This looks good, thanks @soamOne!

**Comment by BenHenning** [comment]
Going ahead and merging this since everything is looking good.

**Pull request merged**